### PR TITLE
#301 Ability to create team slots implemented

### PR DESCRIPTION
**Pull request by YuraRiabov**
Screenshot_888
Screenshot_889

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 1]
```diff
@@ -180,12 +180,17 @@ public async Task<AvailabilitySlotDto> UpdateAvailabilitySlot(long id, SaveAvail
                 .Include(slot => slot.EmailTemplates)
                 .FirstOrDefaultAsync(slot => slot.Id == id) ?? throw new KeyNotFoundException("Availability slot doesn't exist");
 
+            foreach (var member in updateAvailabilityDto.SlotMembers)
+            {
+                member.ScheduleId = availabilitySlot.SlotMembers.First().ScheduleId;
```
Comment (Erbidium) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: maybe we should add check to First() like "member => member.MemberId == availabilitySlot.CreatedBy" which can be seen below

**Review by Erbidium — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 2]
```diff
@@ -180,12 +180,17 @@ public async Task<AvailabilitySlotDto> UpdateAvailabilitySlot(long id, SaveAvail
                 .Include(slot => slot.EmailTemplates)
                 .FirstOrDefaultAsync(slot => slot.Id == id) ?? throw new KeyNotFoundException("Availability slot doesn't exist");
 
+            foreach (var member in updateAvailabilityDto.SlotMembers)
+            {
+                member.ScheduleId = availabilitySlot.SlotMembers.First(m => m.MemberId == availabilitySlot.CreatedBy).ScheduleId;
+            }
+
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Why do we need this? Every member have their own schedule

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 3]
```diff
@@ -259,29 +264,27 @@ private void UpdateQuestions(SaveAvailabilitySlotDto updateAvailabilityDto, Avai
             _context.Update(availabilitySlot);
         }
 
-        private async Task UpdateSchedule(SaveAvailabilitySlotDto updateAvailabilityDto, AvailabilitySlot availabilitySlot)
+        private void UpdateSchedule(SaveAvailabilitySlotDto updateAvailabilityDto, AvailabilitySlot availabilitySlot)
         {
-            var currentUser = await _userService.GetCurrentUserAsync();
-            if (!updateAvailabilityDto.Schedule.WithTeamMembers)
+            var scheduleId = availabilitySlot.SlotMembers.First(member => member.MemberId == availabilitySlot.CreatedBy)
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: see the previous comment

File: frontend/src/app/modules/availability/new-slot/schedule/schedule.component.ts [hunk 4]
```diff
@@ -56,4 +90,51 @@ export class ScheduleComponent extends BaseComponent {
...
+        this.confirmationService.openSlotMembersDialog({
+            alreadySelected: this.slotMembers,
+            teamId: this.teamId,
+            onAddEvent: this.onAddEvent,
+        });
+    }
+
+    public addSlotMembers(members: ISlotMember[]) {
+        this.slotMembers = [...this.slotMembers, ...members];
+    }
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/schedule/schedule.component.ts]: Use one style format (with or without public word)

File: frontend/src/app/shared/components/slot-members-window/slot-members-window.component.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    ) {
+        super();
+        this.alreadySelected = data.alreadySelected;
+        this.teamId = data.teamId;
+        this.teamService.getTeamMembers(this.teamId)
+            .pipe(this.untilThis)
+            .subscribe(members => {
+                this.filteredMembers = members
+                    .map(member => ({ priority: 1, memberId: member.userId, name: member.name, email: member.email, image: member.image }))
+                    .filter(member => !this.alreadySelected.some(m => m.memberId === member.memberId));
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/slot-members-window/slot-members-window.component.ts]: move to a separate method

File: frontend/src/app/shared/components/slot-members-window/slot-members-window.component.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public data: ISlotMemberDialogData,
+        private dialogRef: MatDialogRef<ConfirmationWindowComponent>,
+        private teamService: TeamService,
+    ) {
+        super();
+        this.alreadySelected = data.alreadySelected;
+        this.teamId = data.teamId;
+        this.teamService.getTeamMembers(this.teamId)
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/slot-members-window/slot-members-window.component.ts]: why in the constructor?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

File: frontend/src/app/modules/availability/new-slot/schedule/schedule.component.html [hunk 7]
```diff
@@ -1,18 +1,63 @@
...
+                    <div class="member-wrapper">
+                        <p class="user-name">{{member.name}}</p>
+                        <button mat-icon-button *ngIf="member.memberId !== defaultMember?.memberId" (click)="removeMember(member)">
+                            <mat-icon>close</mat-icon>
+                        </button>
+                    </div>
+                    <mat-divider></mat-divider>
+                </div>
+            </div>
+            <button mat-flat-button (click)="openAddMembers()">+ Add Team Member (s)</button>
```
Comment (apofis96) [inline frontend/src/app/modules/availability/new-slot/schedule/schedule.component.html]: use plus icon

**Review by apofis96 — CHANGES_REQUESTED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**